libappstream-glib/as-app-validate.c

```diff
@@ -1299,6 +1353,42 @@ as_app_validate (AsApp *app, guint32 flags, GError **error)
 		}
 	}
 
+	/* categories */
+	if (as_format_get_kind (format) == AS_FORMAT_KIND_APPSTREAM &&
+	    as_app_get_kind (app) == AS_APP_KIND_DESKTOP) {
+		GPtrArray *categories = as_app_get_categories (app);
+		guint nr_toplevel_cats = 0;
+		const gchar *cats[] = { "AudioVideo",
+					"Development",
+					"Education",
+					"Game",
+					"Graphics",
+					"Network",
+					"Office",
+					"Science",
+					"Settings",
+					"System",
+					"Utility",
+					NULL };
+		for (guint i = 0; i < categories->len; i++) {
+			const gchar *cat = g_ptr_array_index (categories, i);
+			for (guint j = 0; cats[j] != NULL; j++) {
+				if (g_strcmp0 (cats[j], cat) == 0)
+					nr_toplevel_cats++;
+			}
+		}
+		if (nr_toplevel_cats == 0) {
+			ai_app_validate_add (helper,
+					     AS_PROBLEM_KIND_TAG_MISSING,
+					     "main <category> not present");
```

Comment:
I wonder if `<category> must include one of the Main Categories listed in the desktop entry spec` would be a bit clearer?

Reply:
Agreed, thanks.